Check

Pablomediero opened the pull request:
Add Funcionality
Implement RootClass
Control states
Control navigation

Commit:
Check
Add Funcionality
Implement RootClass
Control states
Control navigation

DavidTomas14 commented on app/src/main/java/pmediero/com/features/welcome/presentation/WelcomeScreen.kt:
```diff
@@ -138,8 +144,7 @@ fun BodyWelcomeScreen(modifier: Modifier, navController: NavController) {
                 CustomFloatingActionButtonWithText(
                     modifier =   Modifier.padding(all = LocalSpacing.current.default),
```
val spacing = LocalSpacing.current
mejor que estar repitiendo eso en toda la screen, sácalo a una variable

DavidTomas14 commented on app/src/main/java/pmediero/com/features/addplant/presentation/event/AddPlantEvent.kt:
```diff
@@ -0,0 +1,10 @@
+package pmediero.com.features.addplant.presentation.event
+
+sealed class AddPlantEvent
```
AddPlantAction mejor que event. 
Son las acciones que disparan la pantalla, es mejor naming.
Lo mismo para WelcomeAction y SplashAction

DavidTomas14 commented on app/src/main/java/pmediero/com/features/welcome/presentation/WelcomeRoot.kt:
```diff
@@ -0,0 +1,21 @@
+                navController.popBackStack()
+                navController.navigate(AppRoutes.AddPlantScreen.route)
+            }
+            else -> {}
```
Puedes borrar este else

DavidTomas14 commented on app/src/main/java/pmediero/com/features/welcome/presentation/WelcomeScreen.kt:
```diff
@@ -15,17 +21,17 @@ import androidx.compose.ui.res.painterResource
 
 @Composable
-fun WelcomeScreen(navController: NavController) {
+fun WelcomeScreen(
```
He visto varios textos Hardcodeados es NUNCA NEN. Recuerdame el lunes que te enseñe el UiText o miralo en mi repo, es super util

DavidTomas14 commented on app/src/main/java/pmediero/com/features/addplant/presentation/AddPlantViewModel.kt:
```diff
@@ -0,0 +1,72 @@
+                state = state.copy(
+                    wateringDays = when {
+                        event.wateringDays.all { it.value } -> "EveryDay"
+                        else -> event.wateringDays.filter { it.value }.keys.joinToString {
```
Creo que siendo estrictos con la clean architecture la lógica de negocio va a la capa de domain, es decir a un UseCase. Crea un FilterWateringDaysUseCase

DavidTomas14 commented on app/src/main/java/pmediero/com/features/addplant/presentation/AddPlantViewModel.kt:
```diff
@@ -0,0 +1,72 @@
+                )
+            }
+
+            else -> {}
```
No se usa este else

DavidTomas14 commented on app/src/main/java/pmediero/com/features/addplant/presentation/AddPlantScreen.kt:
```diff
@@ -89,15 +97,36 @@ fun AddPlantScreenFigma() {
                     vertical = LocalSpacing.current.small
-                )
+                ),
+            onEvent = onEvent,
```
```suggestion
            onFooterBtnClick = {
                onEvent(OnCreatePlant)
            },
```

DavidTomas14 commented on app/src/main/java/pmediero/com/features/addplant/presentation/AddPlantScreen.kt:
```diff
@@ -364,19 +417,23 @@ fun FormAddPlantFigma(modifier: Modifier) {
+
 @Composable
-fun FooterAddPlantFigma(modifier: Modifier) {
+fun FooterAddPlantFigma(modifier: Modifier, onEvent: (AddPlantEvent) -> Unit) {
```
```suggestion
ffun FooterAddPlantFigma(modifier: Modifier, onFooterBtnClick: () -> Unit) {
```

DavidTomas14 commented on app/src/main/java/pmediero/com/features/addplant/presentation/AddPlantScreen.kt:
```diff
@@ -364,19 +417,23 @@ fun FormAddPlantFigma(modifier: Modifier) {
     CustomFloatingActionButtonWithText(
         modifier = modifier.fillMaxWidth(),
-        onClick = { /*TODO*/ },
+        onClick = {
+            onEvent(OnCreatePlant)
+        },
```
```suggestion
          onClick = onFooterBtnClick,
```

DavidTomas14 commented on app/src/main/java/pmediero/com/features/addplant/presentation/AddPlantScreen.kt:
```diff
@@ -286,17 +335,17 @@ fun FormAddPlantFigma(modifier: Modifier) {
-                value = wateringDays,
-                onValueChange = { wateringDays = it },
+                value = state.wateringDays,
+                //onValueChange = {  },
```
Quítame los comments CERDO

DavidTomas14 reviewed (changes requested):
Pequeños cambios, en general está muy buen. Buen trabajo!!  😉

Commit:
Check
- UseCase
- StringResource
- LocalSpacing

Commit:
Check
- Implement Koin
- Implement UiText
- Temporal Watering Days Checkbox shortedBy
